types/hexo/index.d.ts

```diff
@@ -0,0 +1,976 @@
+
+            /**
+             * Executed when creating a post to determine the path of new posts.
+             */
+            register(type: 'new_post_path', fn: (data: Post.Data, replace?: boolean) => void, priority?: number): void;
+
+            /**
+             * Used to determine the permalink of posts.
+             */
+            register(type: 'post_permalink', fn: (permalink: string) => string, priority?: number): void;
+
+            /**
+             * Executed after rendering finishes. You can see rendering for more info.
+             */
+            register(type: 'after_render:html', fn: (result: string, data: {path: string, text: string, [key: string]: any}) => string|void, priority?: number): void;
+
+            /**
+             * Executed after generated files and cache are removed with hexo clean command.
+             */
+            register(type: 'after_clean', fn: () => void, priority?: number): void;
+
+            /**
+             * Add middleware to the server. app is a Connect instance.
+             */
+            register(type: 'server_middleware', fn: (app: connect.Server) => connect.Server|void, priority?: number): void;
+
+            unregister(type: string, fn: (...args: any[]) => any): void;
+            exec(type: string, data?: any, options?: Filter.Options): any;
+            execSync(type: string, data?: any, options?: Filter.Options): any;
+        }
+        namespace Filter {
+            interface Options {
+              /**
+               * `hexo` object.
+               */
+              context?: Hexo;
+              /**
+               * Arguments. This must be an array.
+               */
+              args?: any[];
+            }
+        }
+
+        interface Generator {
+            register(name: string, fn: (locals: Site) => Generator.Return| Generator.Return[]): void;
+        }
+        namespace Generator {
+            interface Return {
+              /**
+               * Path not including the prefixing `/` .
+               */
+              path: string;
+
+              /**
+               * Layout. Specify the layouts for rendering. The value can be a string or an array. If it’s ignored then the route will return data directly.
+               */
+              layout: string|string[];
+
+              data: any;
+            }
+        }
+
+        interface Helper {
+            register(name: string, fn: (...args: any[]) => any): void;
+        }
+
+        interface Migrator {
+            register(name: string, fn: (args: ParsedArgs, fn: (err: any) => void) => void): void;
+        }
+
+        interface Processor {
+            register(pattern: RegExp | string | ((str: string) => any), fn: (file: Box.File) => void): void;
+            register(fn: (file: Box.File) => void): void;
+        }
+
+        interface Renderer {
+            register(srcExt: string, outExt: string, fn: (data: HexoRendererData, options?: any) => string, sync: true): void;
```

Comment:
As discussed [here](http://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html), it's not recommended to have optional parameters in callbacks.

Reply:
Thank you!I will fix it.

Reply:
I fixed it. Thank you!